Formula/discrete-scroll.rb

```diff
@@ -0,0 +1,18 @@
+class DiscreteScroll < Formula
+  desc "Fix for OS X's scroll wheel problem"
+  homepage "https://github.com/emreyolcu/discrete-scroll"
+  url "https://github.com/emreyolcu/discrete-scroll/archive/v0.1.1.tar.gz"
+  sha256 "74c94ae4dca1f21a2045fb0700a0ce0e315cf056dea8c7e320a4fc6978c79b03"
+  license "MIT"
+  head "https://github.com/emreyolcu/discrete-scroll.git"
+
+  def install
+    system ENV.cc, "-std=c99", "-O3", "-Wall", "-framework", "Cocoa",
+                   "-o", "discrete-scroll", buildpath/"DiscreteScroll/main.m"
+    bin.install "discrete-scroll" => "discretescroll"
+  end
+
+  test do
+    system "exit"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I don't think this app can be tested.

Reply:
Since it's meant to be a daemon, a bare minimum test would be something like:
1. Run `discretescroll`
1. Sleep 10 seconds
1. Check `ps` output to verify that `discretescroll` is still running
1. Kill `discretescroll`

Also, I just tried building your formula on Catalina, but it just segfaults when I try to run it.

Reply:
It segfaults when it is launched for the first time. An `accessibility access` dialog should appear. Then you should 
grand the permissions from `system preferences > security & privacy > accessibility`. This is how the app is designed. Contrary to its awkward setup routine, it works very well.

Reply:
That's neither obvious nor documented. At least on my Catalina box, the System Preferences dialog pops up _under_ my Terminal window, so I never noticed it until you mentioned it. Also, `discretescroll` is _not_ automatically added to the list of apps; the average user would be left staring at that dialog box and wondering [a] why they were taken there, and [b] what they needed to do.

In contrast, the `DiscreteScroll.app` binary available from the project page Does The Right Thing. Running it pops up a dialog box telling the user that Accessibility permissions need to be added, then when you click on the OK button, the System Preferences dialog pops up _over_ everything else, with DiscreteScroll _already added_ to the Accessibility list of apps.

Given the non-trivial workflow that users would need to understand and implement with this formula, I think packaging `DiscreteScroll.app` as a _cask_ makes more sense.

Reply:
`cask` version is definitely easier to use, but I always prefer compiled version to an unknown binary.

Permission issue is probably because of Cask version being a GUI app, while this one is a command line tool. macOS only shows GUI apps in `accessibility settings` so  `Terminal.app` is shown.

Reply:
I may not be a Homebrew maintainer, but I'm confident that any formula installing binaries that are _expected_ to:
1. Crash on first run
1. Pop-under a system dialog with no explanation why it appeared, and no hints as to what to do next

will not be accepted in `homebrew/core`.

Reply:
yeah, this indeed sounds like a cask formula to me, unless we can figure out a good way to do the test.

Reply:
I guess so. Closing this for now.